src/firmware_new/boards/frankie_v1/BUILD

```diff
@@ -0,0 +1,139 @@
+load("//firmware_new:rules.bzl", "stm32h7_hal_library_files_genrule")
+
+cc_binary(
+    name = "frankie_v1_main.elf",
+    srcs = ["main.c"],
+    copts = [
+        "-DUSE_HAL_DRIVER",
+        "-DSTM32H743xx",
+    ],
+    linkopts = [
+        "-T$(location //firmware_new/boards/frankie_v1:STM32H742XGHx_FLASH.ld)",
+        "-specs=nano.specs",
+        "-lc",
+        "-lm",
+        "-lnosys",
+    ],
+    restricted_to = ["//cc_toolchain:stm32h7"],
+    deps = [
+        ":STM32H742XGHx_FLASH.ld",
+        ":frankie_v1_main_lib",
+    ],
+)
+
+cc_library(
+    name = "frankie_v1_main_lib",
+    srcs = glob(
+        [
+            "*.c",
+            "*.s",
+        ],
+        exclude = ["main.c"],
+    ),
+    hdrs = glob(["*.h"]),
+    copts = [
+        "-DSTM32H743xx",
+    ],
+    restricted_to = ["//cc_toolchain:stm32h7"],
+    deps = [":hal_library"],
+)
+
+stm32h7_hal_library_files_genrule("hal_library_files")
+
+cc_library(
+    name = "hal_library",
+    srcs = [
```

Comment:
Maybe worth commenting that we have do specify all driver files manually, and how to get the list if needed?

Reply:
Added a note:
```
# We got some of this list of hdrs/srcs by looking at the Makefile STM32CubeMX generates,
# (if the project type is set to `MakeFile`) and others by simply attempting compilation,
# seeing where stuff failed, and finding the correct header in the downloaded files
````